src/cycax/cycad/cycad_part.py

```diff
@@ -297,13 +297,18 @@ def insert_hole(self, hole: Holes):
         if self.moves[2] != 0:
             hole.move(z=-self.moves[2])
 
-        rotation = [self.x_size, self.y_size, self.z_size]
-        if self.rotate[0] != 0:
-            rotation = hole.swap_yz(((360 - self.rotate[0]) / 90), rotation)
-        if self.rotate[1] != 0:
-            rotation = hole.swap_xz(((360 - self.rotate[1]) / 90), rotation)
-        if self.rotate[2] != 0:
-            rotation = hole.swap_xy(((360 - self.rotate[2]) / 90), rotation)
+        
+        working_rotate=copy.deepcopy(self.rotate)
+        rotation = [self.x_max-self.x_min, self.y_max-self.y_min, self.z_max-self.z_min]
+        while len(working_rotate)>0:
```

Comment:
from
```
        while len(working_rotate)>0:

            rot=working_rotate.pop()
            if rot==0:
                rotation=hole.swap_yz(3, rotation)
            if rot==1:
                rotation=hole.swap_xz(3, rotation)
            if rot==2:
                rotation=hole.swap_xy(3, rotation)
```
to
```
        for rot in working_rotate:
            if rot==0:
                rotation=hole.swap_yz(3, rotation)
            elif rot==1:
                rotation=hole.swap_xz(3, rotation)
            elif rot==2:
                rotation=hole.swap_xy(3, rotation)
```
For loop just reads better than a while and assign. Also faster, same for elif.